src/main/java/edu/cornell/kfs/module/purap/businessobject/IWantAccount.java

```diff
@@ -207,6 +209,46 @@ public void setUseAmountOrPercent(String useAmountOrPercent) {
     }
 
     /**
+     * Returns the amount-or-percent value if configured to use amounts, zero otherwise.
+     * 
+     * @see org.kuali.kfs.sys.businessobject.GeneralLedgerPendingEntrySourceDetail#getAmount()
+     */
+    @Override
+    public KualiDecimal getAmount() {
+        return CUPurapConstants.AMOUNT.equals(getUseAmountOrPercent()) ? getAmountOrPercent() : KualiDecimal.ZERO;
+    }
+
+    // Only here to comply with the GeneralLedgerPendingEntrySourceDetail interface.
+    @Override
+    public String getBalanceTypeCode() {
+        return null;
+    }
+
+    // Only here to comply with the GeneralLedgerPendingEntrySourceDetail interface.
+    @Override
+    public String getFinancialDocumentLineDescription() {
+        return null;
```

Comment:
I guess these could throw an exception if needed. However, depending on where the IWNT account is used, there could be code that calls these getters, though that may not be the case for Favorite Accounts and the IWNT document purposes. Not sure whether it's good practice for value retrieval methods to throw exceptions when unused.